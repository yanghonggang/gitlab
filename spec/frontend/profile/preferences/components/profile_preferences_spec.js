import { shallowMount } from '@vue/test-utils';

import ProfilePreferences from '~/profile/preferences/components/profile_preferences.vue';
import IntegrationView from '~/profile/preferences/components/integration_view.vue';
import { integrationViews, userFields } from '../mock_data';

describe('ProfilePreferences component', () => {
  let wrapper;
  const defaultProvide = {
    integrationViews: [],
    userFields,
  };

  function createComponent(options = {}) {
    const { props = {}, provide = {} } = options;
    return shallowMount(ProfilePreferences, {
      provide: {
        ...defaultProvide,
        ...provide,
      },
      propsData: props,
    });
  }

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  it('should not render Integrations section', () => {
    wrapper = createComponent();
    const views = wrapper.findAll(IntegrationView);
    const divider = wrapper.find('[data-testid="profile-preferences-integrations-rule"]');
    const heading = wrapper.find('[data-testid="profile-preferences-integrations-heading"]');

    expect(divider.exists()).toBe(false);
    expect(heading.exists()).toBe(false);
    expect(views).toHaveLength(0);
  });

  it('should render Integration section', () => {
    wrapper = createComponent({ provide: { integrationViews } });
    const divider = wrapper.find('[data-testid="profile-preferences-integrations-rule"]');
    const heading = wrapper.find('[data-testid="profile-preferences-integrations-heading"]');
    const views = wrapper.findAll(IntegrationView);

    expect(divider.exists()).toBe(true);
    expect(heading.exists()).toBe(true);
    expect(views).toHaveLength(integrationViews.length);
  });

  it('should render ProfilePreferences properly', () => {
    wrapper = createComponent({ provide: { integrationViews } });

    expect(wrapper.element).toMatchSnapshot();
  });
});
