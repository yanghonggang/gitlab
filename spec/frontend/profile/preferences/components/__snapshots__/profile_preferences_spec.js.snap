// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePreferences component should render ProfilePreferences properly 1`] = `
<div
  class="row gl-mt-3 js-preferences-form"
>
  <div
    class="col-sm-12"
  >
    <hr
      data-testid="profile-preferences-integrations-rule"
    />
  </div>
   
  <div
    class="col-lg-4 profile-settings-sidebar"
  >
    <h4
      class="gl-mt-0"
      data-testid="profile-preferences-integrations-heading"
    >
      
      Integrations
    
    </h4>
     
    <p>
      
      Customize integrations with third party services.
    
    </p>
  </div>
   
  <div
    class="col-lg-8"
  >
    <integration-view-stub
      config="[object Object]"
      helplink="http://foo.com/help"
      message="Click %{linkStart}Foo%{linkEnd}!"
      messageurl="http://foo.com"
    />
    <integration-view-stub
      config="[object Object]"
      helplink="http://bar.com/help"
      message="Click %{linkStart}Bar%{linkEnd}!"
      messageurl="http://bar.com"
    />
  </div>
</div>
`;
